### ISSUE TITLE
Test test_localclient_start.py fails

### ISSUE THREAD

**Issue by mhamra** [wazuh/wazuh#16374]
|wazuh branch | component|
|---|---|
| 4.4| framework/requirements.txt|

## Description
Test test_localclient_start.py fails

After installing 4.4 branch in a "clean" installation with framework's required packages to run unit tests (framework/requirements.txt and jenkins requirements.txt). 

**Note that version 4.5 does not fail**

Environment: ubuntu 20.04, python 3.9 (3.9.16), pip 22.0.4

### Pytest output

```
____________________________________________________________________________________________ test_localclient_start _____________________________________________________________________________________________

pyfuncitem = <Function test_localclient_start>

    def pytest_pyfunc_call(pyfuncitem):  # type: ignore[no-untyped-def]
        """Run coroutines in an event loop instead of a normal function call."""
        fast = pyfuncitem.config.getoption("--aiohttp-fast")
        if asyncio.iscoroutinefunction(pyfuncitem.function):
            existing_loop = pyfuncitem.funcargs.get(
                "proactor_loop"
            ) or pyfuncitem.funcargs.get("loop", None)
            with _runtime_warning_context():
                with _passthrough_loop_context(existing_loop, fast=fast) as _loop:
                    testargs = {
                        arg: pyfuncitem.funcargs[arg]
                        for arg in pyfuncitem._fixtureinfo.argnames
                    }
>                   _loop.run_until_complete(pyfuncitem.obj(**testargs))

/root/wazuh-4.4/lib/python3.9/site-packages/aiohttp/pytest_plugin.py:186:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/lib/python3.9/contextlib.py:126: in __exit__
    next(self.gen)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    @contextlib.contextmanager
    def _runtime_warning_context():  # type: ignore[no-untyped-def]
        """Context manager which checks for RuntimeWarnings.

        This exists specifically to
        avoid "coroutine 'X' was never awaited" warnings being missed.

        If RuntimeWarnings occur in the context a RuntimeError is raised.
        """
        with warnings.catch_warnings(record=True) as _warnings:
            yield
            rw = [
                "{w.filename}:{w.lineno}:{w.message}".format(w=w)
                for w in _warnings
                if w.category == RuntimeWarning
            ]
            if rw:
>               raise RuntimeError(
                    "{} Runtime Warning{},\n{}".format(
                        len(rw), "" if len(rw) == 1 else "s", "\n".join(rw)
                    )
                )
E               RuntimeError: 1 Runtime Warning,
E               /root/wazuh-4.4/lib/python3.9/site-packages/aiohttp/test_utils.py:560:coroutine 'Response.read' was never awaited

/root/wazuh-4.4/lib/python3.9/site-packages/aiohttp/pytest_plugin.py:143: RuntimeError
```

Pytest also generates the following warnings:

```
=============================================================================================== warnings summary ================================================================================================
framework/wazuh/core/utils.py:834
  /root/wazuh-4.4/framework/wazuh/core/utils.py:834: DeprecationWarning: invalid escape sequence \g
    data = re.sub(r'^&backslash;<(.*[^>])$', '&backslash;&lt;\g<1>', data)

lib/python3.9/site-packages/jsonschema/compat.py:6
lib/python3.9/site-packages/jsonschema/compat.py:6
  /root/wazuh-4.4/lib/python3.9/site-packages/jsonschema/compat.py:6: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10 it will stop working
    from collections import MutableMapping, Sequence  # noqa

lib/python3.9/site-packages/aiohttp_jinja2/helpers.py:24
  /root/wazuh-4.4/lib/python3.9/site-packages/aiohttp_jinja2/helpers.py:24: DeprecationWarning: 'contextfunction' is renamed to 'pass_context', the old name will be removed in Jinja 3.1.
    def url_for(

lib/python3.9/site-packages/aiohttp_jinja2/helpers.py:62
  /root/wazuh-4.4/lib/python3.9/site-packages/aiohttp_jinja2/helpers.py:62: DeprecationWarning: 'contextfunction' is renamed to 'pass_context', the old name will be removed in Jinja 3.1.
    def static_url(context: _Context, static_file_path: str) -> str:

lib/python3.9/site-packages/connexion/apis/aiohttp_api.py:48
  /root/wazuh-4.4/lib/python3.9/site-packages/connexion/apis/aiohttp_api.py:48: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def problems_middleware(request, handler):

lib/python3.9/site-packages/connexion/apis/aiohttp_api.py:169
  /root/wazuh-4.4/lib/python3.9/site-packages/connexion/apis/aiohttp_api.py:169: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def _get_openapi_json(self, request):

lib/python3.9/site-packages/connexion/apis/aiohttp_api.py:177
  /root/wazuh-4.4/lib/python3.9/site-packages/connexion/apis/aiohttp_api.py:177: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def _get_openapi_yaml(self, request):

lib/python3.9/site-packages/connexion/apis/aiohttp_api.py:236
  /root/wazuh-4.4/lib/python3.9/site-packages/connexion/apis/aiohttp_api.py:236: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def _get_swagger_ui_home(self, req):

lib/python3.9/site-packages/connexion/apis/aiohttp_api.py:246
  /root/wazuh-4.4/lib/python3.9/site-packages/connexion/apis/aiohttp_api.py:246: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def _get_swagger_ui_config(self, req):

lib/python3.9/site-packages/connexion/apis/aiohttp_api.py:295
  /root/wazuh-4.4/lib/python3.9/site-packages/connexion/apis/aiohttp_api.py:295: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def get_request(cls, req):

lib/python3.9/site-packages/connexion/apis/aiohttp_api.py:324
  /root/wazuh-4.4/lib/python3.9/site-packages/connexion/apis/aiohttp_api.py:324: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def get_response(cls, response, mimetype=None, request=None):

framework/wazuh/core/cluster/tests/test_local_client.py:7
  /root/wazuh-4.4/framework/wazuh/core/cluster/tests/test_local_client.py:7: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10 it will stop working
    from collections import Callable

scripts/tests/test_agent_groups.py::test_show_groups
  /root/wazuh-4.4/framework/scripts/tests/test_agent_groups.py:42: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
    forward_mock.has_calls([call(func=agent.get_agent_groups, kwargs={}),

scripts/tests/test_agent_groups.py::test_show_group
  /root/wazuh-4.4/framework/scripts/tests/test_agent_groups.py:69: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
    forward_mock.has_calls(call(func=agent.get_agents, f_kwargs={'agent_list': agent_id}))

scripts/tests/test_agent_groups.py::test_show_synced_agent
  /root/wazuh-4.4/framework/scripts/tests/test_agent_groups.py:98: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
    forward_mock.has_calls(call(func=agent.get_agents_sync_group, f_kwargs={'agent_list': [agent_id]}))

scripts/tests/test_agent_groups.py::test_show_agents_with_group
  /root/wazuh-4.4/framework/scripts/tests/test_agent_groups.py:125: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
    forward_mock.has_calls(call(func=agent.get_agents_in_group, f_kwargs={'group_list': 'testing'}))

scripts/tests/test_agent_groups.py::test_show_group_files
  /root/wazuh-4.4/framework/scripts/tests/test_agent_groups.py:152: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
    forward_mock.has_calls(call(func=agent.get_group_files, f_kwargs={'group_list': 'testing'}))

scripts/tests/test_agent_groups.py::test_unset_group
  /root/wazuh-4.4/framework/scripts/tests/test_agent_groups.py:181: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
    forward_mock.has_calls(call(func=agent.remove_agent_from_groups,

scripts/tests/test_agent_groups.py::test_remove_group
  /root/wazuh-4.4/framework/scripts/tests/test_agent_groups.py:217: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
    forward_mock.has_calls(call(func=agent.delete_groups, f_kwargs={'group_list': ['testing']}))

scripts/tests/test_agent_groups.py::test_set_group
  /root/wazuh-4.4/framework/scripts/tests/test_agent_groups.py:252: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
    forward_mock.has_calls(call(func=agent.assign_agents_to_group,

scripts/tests/test_agent_groups.py::test_create_group
  /root/wazuh-4.4/framework/scripts/tests/test_agent_groups.py:291: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
    forward_mock.has_calls(call(func=agent.create_group, f_kwargs={'group_id': group_id}))

scripts/tests/test_cluster_control.py::test_main
  /usr/lib/python3.9/unittest/mock.py:328: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
    self.__dict__[_the_name] = value

wazuh/core/cluster/tests/test_client.py::test_ac_connection_lost
  /usr/lib/python3.9/unittest/mock.py:2104: RuntimeWarning: coroutine 'Handler.send_request' was never awaited
    self.name = name

wazuh/core/cluster/tests/test_common.py::test_response_init
  /root/wazuh-4.4/framework/wazuh/core/cluster/tests/test_common.py:66: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    event = asyncio.Event(loop=loop)

wazuh/core/cluster/tests/test_common.py::test_response_read
  /usr/lib/python3.9/unittest/mock.py:2104: RuntimeWarning: coroutine 'test_run_in_pool.<locals>.LoopMock.run_in_executor' was never awaited
    self.name = name

wazuh/core/cluster/tests/test_local_client.py::test_localclient_start
  /root/wazuh-4.4/lib/python3.9/site-packages/pytest_trio/plugin.py:62: TrioDeprecationWarning: trio.MultiError is deprecated since Trio 0.22.0; use BaseExceptionGroup (on Python 3.11 and later) or exceptiongroup.BaseExceptionGroup (earlier versions) instead (https://github.com/python-trio/trio/issues/2211)
    if issubclass(call.excinfo.type, trio.MultiError):

wazuh/core/cluster/tests/test_local_server.py::test_LocalServer_start
  /usr/lib/python3.9/unittest/mock.py:2104: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
    self.name = name

wazuh/core/cluster/tests/test_local_server.py::test_LocalServer_start
  /usr/lib/python3.9/unittest/mock.py:2104: RuntimeWarning: coroutine 'test_LocalServerHandler_get_send_file_response.<locals>.func_mock' was never awaited
    self.name = name

wazuh/core/cluster/tests/test_local_server.py::test_LocalServerHandlerWorker_get_api_response
  /usr/lib/python3.9/tokenize.py:426: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
    return _tokenize(rl_gen.__next__, encoding)

wazuh/core/cluster/tests/test_master.py::test_master_handler_execute_ok
  /usr/lib/python3.9/asyncio/base_events.py:672: RuntimeWarning: coroutine 'test_LocalServerHandlerMaster_send_file_request.<locals>.func_mock' was never awaited
    self._ready.clear()

wazuh/core/cluster/tests/test_master.py::test_master_handler_execute_ko
  /usr/lib/python3.9/unittest/mock.py:2104: RuntimeWarning: coroutine 'Event.wait' was never awaited
    self.name = name

wazuh/core/cluster/tests/test_master.py::test_master_handler_sync_extra_valid
  /usr/lib/python3.9/unittest/mock.py:2104: RuntimeWarning: coroutine 'test_master_handler_sync_worker_files_ko.<locals>.EventMock.wait' was never awaited
    self.name = name

wazuh/core/cluster/tests/test_master.py::test_master_handler_sync_integrity_ok
  /usr/lib/python3.9/unittest/mock.py:2104: RuntimeWarning: coroutine 'test_master_handler_sync_worker_files_ok.<locals>.EventMock.wait' was never awaited
    self.name = name

wazuh/core/cluster/tests/test_master.py::test_master_handler_sync_integrity_ok
  /usr/lib/python3.9/unittest/mock.py:328: RuntimeWarning: coroutine 'test_master_handler_sync_integrity_ok.<locals>.EventMock.wait' was never awaited
    self.__dict__[_the_name] = value

wazuh/core/cluster/tests/test_master.py::test_master_handler_process_files_from_worker_ok
  /usr/lib/python3.9/unittest/mock.py:2058: RuntimeWarning: coroutine 'test_master_handler_sync_integrity_ko.<locals>.EventMock.wait' was never awaited
    setattr(_type, entry, MagicProxy(entry, self))

wazuh/core/cluster/tests/test_master.py::test_master_init
  /usr/lib/python3.9/unittest/mock.py:1074: RuntimeWarning: coroutine 'test_master_handler_sync_integrity_ok.<locals>.EventMock.wait' was never awaited
    _safe_super(CallableMixin, self).__init__(

wazuh/core/cluster/tests/test_master.py::test_master_init
  /usr/lib/python3.9/unittest/mock.py:1074: RuntimeWarning: coroutine 'test_master_handler_sync_integrity_ko.<locals>.EventMock.wait' was never awaited
    _safe_super(CallableMixin, self).__init__(

wazuh/core/cluster/tests/test_worker.py::test_rit_done_callback
wazuh/core/cluster/tests/test_worker.py::test_worker_handler_process_request_ok
  /usr/lib/python3.9/unittest/mock.py:2058: RuntimeWarning: coroutine 'WorkerHandler.process_files_from_master' was never awaited
    setattr(_type, entry, MagicProxy(entry, self))

wazuh/core/tests/test_agent.py::test_agent_set_agent_group_relationship[False-False-append]
  /usr/lib/python3.9/unittest/mock.py:1074: RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
    _safe_super(CallableMixin, self).__init__(

wazuh/core/tests/test_agent.py::test_agent_set_agent_group_relationship[False-False-append]
  /usr/lib/python3.9/unittest/mock.py:1074: RuntimeWarning: coroutine 'AbstractServer.check_clients_keepalive' was never awaited
    _safe_super(CallableMixin, self).__init__(

wazuh/rbac/tests/test_auth_context.py::test_auth_roles
  /root/wazuh-4.4/framework/wazuh/rbac/auth_context.py:177: FutureWarning: Possible nested set at position 2
    regex = re.compile(regex)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
```

**Comment by mhamra** [comment]
### UPDATE
Results obtained setting PYTHONPATH=[path_to_repo]/api:[path_to_repo]/framework variable and running pytest on framework/wazuh/core/cluster/tests/ folder:

- test_local_client.py::test_localclient_start fails when all tests in the folder are run without changing the alphabetical order found in the repository
- Running only the tests inside "test_local_client.py" makes no failure
- Changing the alphabetical order of the files inside "framework/wazuh/core/cluster/tests/" (by changing file names) and running all tests generally makes no failure on any test of the folder
- If the test file "test_cluster.py" is moved right before the file "test_worker.py" and all tests on the folder are run, the test  "test_cluster.py::test_decompress_files_ok" fails

In all of the cases that a test fails, the problem arises because the function pytest_plugin::_runtime_warning_context from the aiohttp module catches a warning about "coroutine 'X' was never awaited" and raises an exception. For example:

- When test_cluster::test_decomplress_files_ok fails
`
E               /usr/lib/python3.9/unittest/mock.py:2104:coroutine 'AsyncMockMixin._execute_mock_call' was never awaited
E               /usr/lib/python3.9/unittest/mock.py:2104:coroutine 'AbstractServer.check_clients_keepalive' was never awaited
`
- when local_cliente::test_localclient_start failse
`
E               /root/wazuh-4.4/lib/python3.9/site-packages/aiohttp/test_utils.py:568:coroutine 'Response.read' was never awaited
`

Both cases seem to be related to testing libraries (test_client.py from aiohttp and mock.py from unittest. The errors are not raisen inside test code, they seem to be raisen once executed.

**Comment by mhamra** [comment]
### UPDATE
Upgrading current packages to newer ones

current | updated
-- | --
hiredis==1.1.0 | hiredis==2.2.2
pytest==5.4.3 | pytest==7.0.1
pytest-aiohttp==0.3.0 | pytest-aiohttp==1.0.4
pytest-asyncio==0.15.1 | pytest-asyncio==0.18.1

All tests are passed successfully

**Comment by mhamra** [comment]
This issue is addressed by [#5039 - Bump test dependencies versions to solve python 3.10 tests warnings](https://github.com/wazuh/wazuh-jenkins/issues/5039) as part of the epic [#13265 - Upgrade the Wazuh embedded Python to 3.10](https://github.com/wazuh/wazuh/issues/13265)

**Comment by EduLeon12** [comment]
## Review Update

After testing the issue the error was easily replicated and after upgrading the packages mentioned [here](https://github.com/wazuh/wazuh/issues/16374#issuecomment-1471812511) one by one, it was found the _pytest_asyncio_ package wasn't necessary to upgrade to make the test passed but it was necessary to stop some test from being skipped and reduce the warnings. Therefore the issue scope has been achieved and it's approved.

LGTM !!
